compiler/mio-tflite260/src/Helper.cpp

```diff
@@ -23,22 +24,52 @@ namespace mio
 namespace tflite
 {
 
-// This will provide v3/v3a format neutral BuiltinOperator
+/**
+ * This will provide v3/v3a format neutral BuiltinOperator
+ *
+ * This function referenced
+ * https://github.com/tensorflow/tensorflow/blob/7d12007d7800d3714a02e05059f3ea602d1aec78/tensorflow/lite/schema/schema_utils.cc
+ * and
+ * https://github.com/tensorflow/tensorflow/commit/cf28969fb2ea72c9738cd3acdddc0b69fd7dff5e
+ */
 ::tflite::BuiltinOperator builtin_code_neutral(const ::tflite::OperatorCode *opcode)
 {
   assert(opcode != nullptr);
-  int8_t dp_code = opcode->deprecated_builtin_code();
-  // 127 is max of int8_t which is upper bound of v3 builtin_code
-  // NOTE TensorFlow uses 'BuiltinOperator_PLACEHOLDER_FOR_GREATER_OP_CODES' for 127
-  if (dp_code < 127 && dp_code >= 0)
-    return ::tflite::BuiltinOperator(dp_code);
-  return opcode->builtin_code();
+  return std::max(opcode->builtin_code(),
+                  static_cast<::tflite::BuiltinOperator>(opcode->deprecated_builtin_code()));
 }
 
+/**
+ * This function does not validate the relation between deprecated_builtin_code() and
+ * builtin_code() because information is not
+ * i.e., which value is stored when builtin_code() is larger than
+ * BuiltinOperator_PLACEHOLDER_FOR_GREATER_OP_CODES?
+ *
+ * According to
+ * https://github.com/tensorflow/tflite-micro/blob/25da8245417b834ced53df722d6a0f517eb425b4/tensorflow/lite/micro/test_helpers.cc#L154,
+ * 0 is stored.
+ *
+ * According to
+ * https://github.com/tensorflow/tensorflow/blob/1ab788fa8d08430be239ab970980b891ad7af494/tensorflow/lite/schema/schema_conversion_utils.cc#L46-L50,
+ * 127 is stored.
+ *
+ * According to schema.fbs of TFLite,
+ * any information is found.
+ *
+ * Until it becomes clear, do not validate about it.
+ */
```

Comment:
can you please elaborate this?

Reply:
I agree it's little bit dirty but I had no idea for long URL T.T
Except long URL, which point seems to be elaborated? :D

Reply:
I can't understand the point of explanation. especially what do you mean by `Until it becomes clear, do not validate about it.` ?

Reply:
I see it :D
Then it would be better to remove this comment and then add it later if related codes are written :)